new-installer/internal/steps/common/argocd_setup.go

```
@@ -0,0 +1,283 @@
+	return &ArgoCDStep{
+		RootPath:               rootPath,
+		KeepGeneratedFiles:     keepGeneratedFiles,
+		orchConfigReaderWriter: orchConfigReaderWriter,
+	}
+}
+
+func (s *ArgoCDStep) Name() string {
+	return "GenericStep"
+}
+
+func (s *ArgoCDStep) Labels() []string {
+	return s.StepLabels
+}
+
+func (s *ArgoCDStep) ConfigStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+
+	return runtimeState, nil
+}
+
+func (s *ArgoCDStep) PreStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+	// no-op for now
+	argocdValues(config)
+	return runtimeState, nil
+}
+
+func (s *ArgoCDStep) RunStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+	// InstallArgoCD
+	err := InstallArgoCD()
+	// if err != nil {
+	// 	// fmt.Printf("Error installing Argocd %v \n", err)
+	// }
+
+	if err != nil {
+		return runtimeState, &internal.OrchInstallerError{
+			ErrorCode: internal.OrchInstallerErrorCodeInternal,
+			ErrorMsg:  fmt.Sprintf("Error installing Argocd %v \n", err),
+		}
+	}
+	return runtimeState, nil
+}
+
+func (s *ArgoCDStep) PostStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState, prevStepError *internal.OrchInstallerError) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+	// Wait for ArgoCD namespace creation normally taken from an env var
+	argoCDNS := "argocd"
+	if argoCDNS == "" {
+		argoCDNS = "argocd"
+	}
+	err := WaitForNamespaceCreation(argoCDNS)
+	if err != nil {
+		// return nil,fmt.Errorf("failed to wait for ArgoCD namespace: %v", err)
+		fmt.Printf("failed to wait for ArgoCD namespace: %v\n", err)
```

commented code, should probably return the error?